### ISSUE TITLE
Issue with Kohya-style LoRA and SD1.5 based model: Weird Images

### ISSUE THREAD

**Issue by antonioglass** [huggingface/diffusers#4668]
### Describe the bug

Weird images are being generated with Kohya-style LoRA and SD1.5 based model (Deliberate 2.0). It works well with A1111. I also tried to run it with kohya_lora_loader (https://gist.github.com/takuma104/e38d683d72b1e448b8d9b3835f7cfa44) and the results are much better.
![download](https://github.com/huggingface/diffusers/assets/71437038/a7e9a8a2-f61f-4fe4-ba27-9d042b7406c8)

### Reproduction

Run image generation according to Huggingface docs: https://huggingface.co/docs/diffusers/training/lora#supporting-a1111-themed-lora-checkpoints-from-diffusers

### System Info

diffusers 0.20.0

### Who can help?

@williamberman, @patrickvonplaten, @sayakpaul

**Comment by sayakpaul** [comment]
Need a fully reproducible snippet.

**Comment by antonioglass** [comment]
Sorry, here it is. I tried both, original SD 1.5 and Deliberate 2.0 and there's the issue in both cases.

```
import torch

from diffusers import StableDiffusionPipeline, DPMSolverMultistepScheduler

pipeline = StableDiffusionPipeline.from_pretrained(
    "runwayml/stable-diffusion-v1-5", torch_dtype=torch.float16, safety_checker=None, use_safetensors=True
).to("cuda")
pipeline.scheduler = DPMSolverMultistepScheduler.from_config(
    pipeline.scheduler.config, use_karras_sigmas=True
)

pipeline.load_lora_weights(".", weight_name="asigi.safetensors")

prompt = "asigi"
negative_prompt = "(deformed, distorted, disfigured)1.3, poorly drawn, bad anatomy, wrong anatomy, extra limb, missing limb, floating limbs, (mutated hands and fingers)1.4, disconnected limbs, mutation, mutated, ugly, disgusting, blurry, amputation"

image = pipeline(prompt=prompt, 
    negative_prompt=negative_prompt, 
    width=512, 
    height=768, 
    num_inference_steps=22, 
    num_images_per_prompt=1,
    generator=torch.manual_seed(838383)
).images[0]

image.save("output.png")
```

**Comment by sayakpaul** [comment]
And the same settings with https://gist.github.com/takuma104/e38d683d72b1e448b8d9b3835f7cfa44 refers to better results? 

Also, I don't see the safetensors files. Could you upload them on the HF Hub platform?

**Comment by antonioglass** [comment]
Yes, it's much better but still not as good as with A1111. You can get a LoRa safetensors from here: https://huggingface.co/antonioglass/asigi/blob/main/asigi.safetensors

Have you tested any Kohya LoRa with SD 1.5 based models? As I understand, there are different types of LoRa's and only some of them are supported in diffusers, correct?

Thank you!

**Comment by sayakpaul** [comment]
The bad quality could be attributed to prompt weighting maybe because the prompt you're using seems to be indicative of you're using prompt weighting in A1111 (I could be wrong).

**Comment by antonioglass** [comment]
Nope, it's not the reason, and it doesn't work in this code as compel is not being engaged. I tried it without the prompt weightning too. Would you have a chance to look into it? I would really appreciate that.

**Comment by sayakpaul** [comment]
We did try a couple of SD v1. 5 LoRA actually and it worked as expected. If there's any unsupported ones, there would have been errors. So, right off the bat I don't have anything concrete to offer. 

Ccing @isidentical and @takuma104 if they have anything further to add.

**Comment by antonioglass** [comment]
Is it possible to identify the type of LoRa based on a safetensors file only? What types do you support?
